bfabric/bfabric.py

```diff
@@ -84,33 +84,58 @@ class Bfabric(object):
     def warning(self, msg):
         sys.stderr.write("\033[93m{}\033[0m\n".format(msg))
 
-    def __init__(self, login=None, password=None, webbase=None, externaljobid=None, bfabricrc=None, verbose=False):
+    def __init__(self, login: str = None, password: str = None, webbase: str = None, externaljobid=None,
+                 config_path: str = None, config_env: str = None, optional_auth: bool = False, verbose: bool = False):
+        """
+        :param login:           Login string for overriding config file
+        :param password:        Password for overriding config file
+        :param webbase:         Webbase for overriding config file
+        :param externaljobid:   ?
+        :param config_path:     Path to the config file, in case it is different from default
+        :param config_env:      Which config environment to use. Can also specify via environment variable or use
+           default in the config file (at your own risk)
+        :param optional_auth:   Whether authentification is optional. If yes, missing authentification will be ignored,
+           otherwise an exception will be raised
+        :param verbose:         Verbosity (TODO: resolve potential redundancy with logger)
+        """
         self.verbose = verbose
 
         self.cl = {}
         self.verbose = False
         self.query_counter = 0
 
-        bfabricrc = bfabricrc or os.path.normpath(os.path.expanduser("~/.bfabricrc.py"))
-        if not os.path.isfile(bfabricrc):
-            self.warning("could not find '.bfabricrc.py' file in home directory.")
-            self.config = BfabricConfig(base_url=webbase)
+        # Get default path config file path
+        config_path = config_path or os.path.normpath(os.path.expanduser("~/.bfabricpy.yml"))
+
+        # Use the provided config data from arguments instead of the file
+        if not os.path.isfile(config_path):
+            self.warning("could not find '.bfabricpy.yml' file in home directory.")
```

Comment:
As I already said, I'm not fully sure when this should go to main yet...

However, once we do decide to release it, maybe it would make sense to add a check here that essentially raises an error if `~/.bfabricrc.py` exists, with a short comment that it should be migrated to `.yml`? During development it could be annoying I guess.